### PR TITLE
Fix for uninitialised console and yast bootloader

### PR DESCRIPTION
**Pull request by gmoro**

### CODE REVIEW

File: files/usr/lib/jeos-firstboot [hunk 1]
```diff
@@ -30,6 +30,12 @@ EULA_FILE=/etc/YaST2/licenses/base/license.txt
 
 stty_size() {
     set -- `stty size`; LINES=$1; COLUMNS=$2
+    # stty size can return zero when not ready or
+    # its a serial console
+    if [ "${COLUMNS}" = "0" || "${LINES}" = "0" ]; then
```
Comment (lnussel) [inline files/usr/lib/jeos-firstboot]: curly brackets are not need here. Use -o instead of ||

File: files/usr/lib/jeos-firstboot [hunk 2]
```diff
@@ -220,6 +226,15 @@ if [ -n "$password" ]; then
     echo "root:$password" | run /usr/sbin/chpasswd
 fi
 
+EFIVARS_DIR="/sys/firmware/efi"
+# Look for EFI dir to see if the machine is booted in UEFI mode
+modprobe efivars
+if ! [ "$(ls -A $EFIVARS_DIR)" ]; then
```
Comment (lnussel) [inline files/usr/lib/jeos-firstboot]: Raymund recommends to check for /sys/firmware/efi/systab, that's also what yast does.

File: files/usr/lib/jeos-firstboot [hunk 3]
```diff
@@ -220,6 +226,15 @@ if [ -n "$password" ]; then
...
+    sed -i -e 's/LOADER_TYPE=.*/LOADER_TYPE=grub2/g' /etc/sysconfig/bootloader 
+else
+    echo "Grub is configured for EFI"
```
Comment (lnussel) [inline files/usr/lib/jeos-firstboot]: useless message IMO, just omit it

File: files/usr/lib/jeos-firstboot [hunk 4]
```diff
@@ -220,6 +226,15 @@ if [ -n "$password" ]; then
     echo "root:$password" | run /usr/sbin/chpasswd
 fi
 
+EFIVARS_DIR="/sys/firmware/efi"
+# Look for EFI dir to see if the machine is booted in UEFI mode
+modprobe efivars
+if ! [ "$(ls -A $EFIVARS_DIR)" ]; then
+    sed -i -e 's/LOADER_TYPE=.*/LOADER_TYPE=grub2/g' /etc/sysconfig/bootloader
```
Comment (lnussel) [inline files/usr/lib/jeos-firstboot]: run sed ... to make dry mode work

File: files/usr/lib/systemd/system/jeos-firstboot.service [hunk 5]
```diff
@@ -11,6 +11,7 @@ DefaultDependencies=no
 Conflicts=shutdown.target
 After=systemd-remount-fs.service
 Before=systemd-sysusers.service sysinit.target shutdown.target
+Wants=systemd-udev-settle.service
```
Comment (lnussel) [inline files/usr/lib/systemd/system/jeos-firstboot.service]: before of cycles here. those deps have been tuned back and forth several times.

File: files/usr/lib/jeos-firstboot (hunk 1 above)
Comment (gmoro) [inline files/usr/lib/jeos-firstboot]: ok, fixed

File: files/usr/lib/jeos-firstboot (hunk 2 above)
Comment (gmoro) [inline files/usr/lib/jeos-firstboot]: yeah, I looked again in the YasT code, was going with a comment somewhere saying if the dir is not empty is enough to know, but better to keep in sync with YasT implementation that is well thought

File: files/usr/lib/jeos-firstboot (hunk 4 above)
Comment (gmoro) [inline files/usr/lib/jeos-firstboot]: fixed

File: files/usr/lib/jeos-firstboot (hunk 3 above)
Comment (gmoro) [inline files/usr/lib/jeos-firstboot]: indeed, did that for the test and forgot to take it out

File: files/usr/lib/systemd/system/jeos-firstboot.service (hunk 5 above)
Comment (gmoro) [inline files/usr/lib/systemd/system/jeos-firstboot.service]: added that a long time ago, still not sure if it's needed, the problem happened to be another thing, will remove for now

**Comment by gmoro** [comment]
second round :)

**Pull request merged**